### PR TITLE
Initialize genesis block with cashed hash

### PR DESCRIPTION
**Pull request by fiaxh**
## Pull request checklist

- [ ] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

**Review by nibhar — APPROVED** [review]

File: genesis/src/networks.rs [hunk 1]
```diff
@@ -41,8 +41,11 @@ impl NetworkInfo {
 
     #[inline]
     pub fn genesis_block(&self) -> Block {
-        Block::deserialize_from_vec(self.genesis.block)
-            .expect("Failed to deserialize genesis block.")
+        let mut block = Block::deserialize_from_vec(self.genesis.block)
+            .expect("Failed to deserialize genesis block.");
+        let mut macro_block = block.unwrap_macro_ref_mut();
+        macro_block.header.cached_hash = Some(self.genesis.hash.clone());
```
Comment (hrxi) [inline genesis/src/networks.rs]:
```suggestion
        block.unwrap_macro_ref_mut().header.cached_hash = Some(self.genesis.hash.clone());
```

**Review by jsdanielh — APPROVED** [review]
LGTM

**Review by sisou — APPROVED** [review]
Tested in WASM, works!

**Review by paberr — APPROVED** [review]
LGTM

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**